### PR TITLE
fix: Suppress extra indent after the end of field and function chains

### PR DESCRIPTION
**Pull request by DropDemBits**
(spurred on by <https://github.com/rust-lang/rust-analyzer/issues/4182#issuecomment-671275652>)

Caveat that this doesn't work for after tail expressions, although there shouldn't be anything after those anyways.

This also complicates when to reload the language configuration by nature of now always having a language configuration applicable.

Examples of indentation fixes:

```rs
fn main() {
    println!("Hello!"); // < enter here!
    // ... indents down here

    fs::read_to_string("soup") // < enter here!
    // ... still indents down here :(
        .map(|_| ())
        .map(|_| ()) // < enter here!
        // ... still indents down here :D
        .map_err(|_| ())
        .unwrap(); // < enter here!
    // ... indents down here :D

    // ... and subsequent enters stay at the same indent

    0.0f64
        .to_radians()
        .to_radians()
        .to_radians() // force semi on a new line
        ; // < enter here!
    // ... indents down here :D
}

fn tail_end() -> i32 {
    0i32.wrapping_abs()
        .wrapping_abs()
        .wrapping_abs()
        .wrapping_abs() // < enter here!
        // ... still indents here :shrug:
}
```

### CODE REVIEW

**Comment by Veykril** [comment]
I think we can define these rules statically here instead? https://github.com/rust-lang/rust-analyzer/blob/master/editors/code/language-configuration.json

**Comment by DropDemBits** [comment]
Oh, I forgot to mention that I tried to add it in the language config file first, but because the onEnter rules are dynamically (re)loaded to include/exclude the comment onEnter rules, the static configuration file isn't used (or at least, that's my best guess as to what's happening).

File: editors/code/src/config.ts [hunk 1]
```diff
@@ -86,58 +88,91 @@ export class Config {
      * [1]: https://github.com/Microsoft/vscode/issues/11514#issuecomment-244707076
      */
     private configureLanguage() {
-        if (this.typingContinueCommentsOnNewline && !this.configureLang) {
+        // Only need to dispose of the config if there's a change
+        if (
+            this.configureLang &&
+            this.typingContinueCommentsOnNewline !==
+                this.configureLang.typingContinueCommentsOnNewline
```
Comment (Veykril) [inline editors/code/src/config.ts:95]: I don't think we need this check, it's fine to reload the on enter stuff whenever this is called now

**Review by Veykril — COMMENTED** [review]
Sounds reasonable that this might override the static configuration there
@bors delegate+

File: editors/code/src/config.ts (hunk 1 above)
Comment (DropDemBits) [reply editors/code/src/config.ts]: Removed :+1:

**Pull request merged**